test/jdk/java/lang/invoke/defineHiddenClass/BasicTest.java

```diff
@@ -263,46 +269,17 @@ private Object[][] emptyClasses() {
      * class.
      */
     @Test(dataProvider = "emptyClasses")
-    public void emptyHiddenClass(String name, int accessFlags) throws Exception {
-        byte[] bytes = (accessFlags == ACC_ENUM) ? classBytes(name, Enum.class, accessFlags)
+    public void emptyHiddenClass(String name, Set<AccessFlag> accessFlags) throws Exception {
+        byte[] bytes = (accessFlags.equals(Set.of(ENUM))) ? classBytes(name, CD_Enum, accessFlags)
                                                  : classBytes(name, accessFlags);
         Class<?> hc = lookup().defineHiddenClass(bytes, false).lookupClass();
-        switch (accessFlags) {
-            case ACC_SYNTHETIC:
-                assertTrue(hc.isSynthetic());
-                assertFalse(hc.isEnum());
-                assertFalse(hc.isAnnotation());
-                assertFalse(hc.isInterface());
-                break;
-            case ACC_ENUM:
-                assertFalse(hc.isSynthetic());
-                assertTrue(hc.isEnum());
-                assertFalse(hc.isAnnotation());
-                assertFalse(hc.isInterface());
-                break;
-            case ACC_ABSTRACT:
-                assertFalse(hc.isSynthetic());
-                assertFalse(hc.isEnum());
-                assertFalse(hc.isAnnotation());
-                assertFalse(hc.isInterface());
-                break;
-            case ACC_ABSTRACT|ACC_INTERFACE:
-                assertFalse(hc.isSynthetic());
-                assertFalse(hc.isEnum());
-                assertFalse(hc.isAnnotation());
-                assertTrue(hc.isInterface());
-                break;
-            case ACC_ANNOTATION|ACC_ABSTRACT|ACC_INTERFACE:
-                assertFalse(hc.isSynthetic());
-                assertFalse(hc.isEnum());
-                assertTrue(hc.isAnnotation());
-                assertTrue(hc.isInterface());
-                break;
-            default:
-                throw new IllegalArgumentException("unexpected access flag: " + accessFlags);
-        }
+        assertEquals(hc.isSynthetic(), accessFlags.contains(SYNTHETIC));
+        assertEquals(hc.isEnum(), accessFlags.contains(ENUM));
+        assertEquals(hc.isAnnotation(), accessFlags.contains(ANNOTATION));
+        assertEquals(hc.isInterface(), accessFlags.contains(INTERFACE));
+
         assertTrue(hc.isHidden());
-        assertTrue(hc.getModifiers() == (ACC_PUBLIC|accessFlags));
+        assertEquals(hc.getModifiers(), accessFlags.stream().mapToInt(AccessFlag::mask).reduce(AccessFlag.PUBLIC.mask(), (a, b) -> a | b));
```

Comment:
Do you mind splitting this line too, it's 140+ characters long so impossible to look at the changes side-by-side.